packages/eslint-plugin-template/tests/rules/accessibility-alt-text.test.ts

```diff
@@ -0,0 +1,73 @@
+import {
+  convertAnnotatedSourceToFailureCase,
+  RuleTester,
+} from '@angular-eslint/utils';
+import rule, {
+  MessageIds,
+  RULE_NAME,
+} from '../../src/rules/accessibility-alt-text';
+
+//------------------------------------------------------------------------------
+// Tests
+//------------------------------------------------------------------------------
+
+const ruleTester = new RuleTester({
+  parser: '@angular-eslint/template-parser',
+});
+
+const messageId: MessageIds = 'accessibilityAltText';
+
+ruleTester.run(RULE_NAME, rule, {
+  valid: [
+    '<img src="foo" alt="Foo eating a sandwich.">',
+    '<img src="foo" [attr.alt]="altText">',
+    `<img src="foo" [attr.alt]="'Alt Text'">`,
+    '<img src="foo" alt="">',
+    '<object aria-label="foo">',
+    '<object aria-labelledby="id1">',
+    '<object>Meaningful description</object>',
+    '<object title="An object">',
+    '<area aria-label="foo"></area>',
+    '<area aria-labelledby="id1"></area>',
+    '<area alt="This is descriptive!"></area>',
+    '<input type="text">',
+    '<input type="image" alt="This is descriptive!">',
+    '<input type="image" aria-label="foo">',
+    '<input type="image" aria-labelledby="id1">',
+  ],
+  invalid: [
+    convertAnnotatedSourceToFailureCase({
+      messageId,
+      description: 'should fail image does not have alt text',
+      annotatedSource: `
+        <img src="foo">
```

Comment:
@JamesHenry 

Hey James. I've got a question. The end source span is **not** set for void elements. What is the proper way to get the _location_ of this node?

Reply:
I’d be curious if it was there if you added the / in front of the > denoting it is the closing tag of a void element...

In any case though questions about the AST the angular compiler produces for us kind of have to be posted on their repo. Most things I know about it have just been inferred by fiddling with and logging the output of things, and fortunately that’s been enough most of the time.

Reply:
I've added a feature to get the location of void elements. Could you have a look, please.

Reply:
Nice work, this seems reasonable to me!